src/Examples/Flowsheets/HDA_flowsheet_with_costing.ipynb

```diff
@@ -0,0 +1,1598 @@
+    "\n",
+    "To demonstrate proper usage, the Compressor costing blocks are defined below. For this example, the compressors assume isothermal thermodynamics and are too simple for the costing module. As shown below, attempting to cost the compressors throws a `ValueError` warning:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2022-07-21 10:58:20 [WARNING] idaes.models.costing.SSLW: fs.C101 - pressure changers without isentropic assumption are too simple to be costed.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# costing for compressors - m.fs.C101, n.fs.C101\n",
+    "\n",
+    "m.fs.C101.costing = UnitModelCostingBlock(\n",
+    "    default={\n",
+    "        \"flowsheet_costing_block\": m.fs.costing,\n",
+    "        \"costing_method\": SSLWCostingData.cost_compressor,\n",
+    "        \"costing_method_arguments\": {\n",
+    "            \"compressor_type\": CompressorType.Centrifugal,\n",
+    "            \"drive_type\": CompressorDriveType.ElectricMotor,\n",
+    "            \"material_type\": CompressorMaterial.StainlessSteel,\n",
+    "        },\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "# since this compressor is too simple to cost, remove costing block\n",
+    "delattr(m.fs.C101, \"costing\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2022-07-21 10:58:20 [WARNING] idaes.models.costing.SSLW: fs.C101 - pressure changers without isentropic assumption are too simple to be costed.\n"
+     ]
+    }
+   ],
+   "source": [
+    "n.fs.C101.costing = UnitModelCostingBlock(\n",
+    "    default={\n",
+    "        \"flowsheet_costing_block\": n.fs.costing,\n",
+    "        \"costing_method\": SSLWCostingData.cost_compressor,\n",
+    "        \"costing_method_arguments\": {\n",
+    "            \"compressor_type\": CompressorType.Centrifugal,\n",
+    "            \"drive_type\": CompressorDriveType.ElectricMotor,\n",
+    "            \"material_type\": CompressorMaterial.StainlessSteel,\n",
+    "        },\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "# since this compressor is too simple to cost, remove costing block\n",
+    "delattr(n.fs.C101, \"costing\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To demonstrate the flexibility of the framework, let's define all Heater costing blocks sequentially using a loop:"
```

Comment:
Give a single, concrete example first (no loops are anything else to obscure things. I would suggest starting with the reactor, especially seeing as it needs additional variables added. Be sure to explain what each argument is for.

You can then start doing other unit types in groups, but make sure to highlight that different unit types have different arguments,